javatests/arcs/android/storage/database/DatabaseImplTest.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.database
+
+import android.database.Cursor
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.storage.database.DatabaseImpl
+import arcs.core.data.*
+import arcs.core.testutil.assertThrows
+import com.google.common.truth.Truth.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.lang.IllegalArgumentException
+
+@RunWith(AndroidJUnit4::class)
+class DatabaseImplTest {
+    private lateinit var database: DatabaseImpl
+
+    /** The first free Type ID after all primitive types have been assigned. */
+    private val FIRST_ENTITY_TYPE_ID = 3
+
+    @Before
+    fun setUp() {
+        database = DatabaseImpl(ApplicationProvider.getApplicationContext())
+    }
+
+    @After
+    fun tearDown() {
+        database.reset()
+    }
+
+    @Test
+    fun getTypeId_primitiveTypeIds() {
+        assertThat(database.getTypeId(FieldType.Boolean)).isEqualTo(PrimitiveType.Boolean.ordinal)
+        assertThat(database.getTypeId(FieldType.Number)).isEqualTo(PrimitiveType.Number.ordinal)
+        assertThat(database.getTypeId(FieldType.Text)).isEqualTo(PrimitiveType.Text.ordinal)
+    }
+
+    @Test
+    fun getTypeId_entity_throwsWhenMissing() {
+        val exception = assertThrows(IllegalArgumentException::class) {
+            database.getTypeId(FieldType.EntityRef("abc"))
+        }
+        assertThat(exception)
+            .hasMessageThat()
+            .isEqualTo("Unknown type ID for schema with hash abc")
+    }
+
+    @Test
+    fun getOrCreateSchemaTypeId_newSchema() {
+        val schema = newSchema("abc")
+
+        assertThat(database.getOrCreateSchemaTypeId(schema)).isEqualTo(FIRST_ENTITY_TYPE_ID)
```

Comment:
Since we're almost always going to be "gettingOrCreating" SchemaTypeId  (I think this is accurate?), maybe this method name can be simplified to `getSchemaTypeId`, and just make it clear in the KDoc that it will be created if it doesn't exist yet.

Reply:
:ok_hand: